experimental/packages/opentelemetry-sdk-metrics-base/src/state/MeterSharedState.ts

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { HrTime } from '@opentelemetry/api';
+import * as metrics from '@opentelemetry/api-metrics-wip';
+import { InstrumentationLibrary } from '@opentelemetry/core';
+import { MetricData } from '..';
+import { createInstrumentDescriptorWithView, InstrumentDescriptor } from '../InstrumentDescriptor';
+import { isNotNullish, promiseFinally } from '../utils';
+import { AsyncMetricStorage } from './AsyncMetricStorage';
+import { MeterProviderSharedState } from './MeterProviderSharedState';
+import { MetricCollectorHandle } from './MetricCollector';
+import { MetricStorage } from './MetricStorage';
+import { MultiMetricStorage } from './MultiWritableMetricStorage';
+import { ObservableRegistry } from './ObservableRegistry';
+import { SyncMetricStorage } from './SyncMetricStorage';
+
+/**
+ * An internal record for shared meter provider states.
+ */
+export class MeterSharedState {
+  private _metricStorageRegistry = new Map<string, MetricStorage>();
+  private _pendingCollectPromise: Promise<MetricData[]> | null = null;
+  private _observableRegistry = new ObservableRegistry();
+
+  constructor(private _meterProviderSharedState: MeterProviderSharedState, private _instrumentationLibrary: InstrumentationLibrary) {}
+
+  registerMetricStorage(descriptor: InstrumentDescriptor) {
+    const views = this._meterProviderSharedState.viewRegistry.findViews(descriptor, this._instrumentationLibrary);
+    const storages = views.map(view => {
+      const viewDescriptor = createInstrumentDescriptorWithView(view, descriptor);
+      const aggregator = view.aggregation.createAggregator(viewDescriptor);
+      const storage = new SyncMetricStorage(viewDescriptor, aggregator, view.attributesProcessor);
+      // TODO: handle conflicts
```

Comment:
Is there an issue for this TODO so it doesn't get forgotten

Reply:
It is being worked at https://github.com/open-telemetry/opentelemetry-js/pull/2734